ReferenceAdapter/src/main/java/com/chartboost/mediation/referenceadapter/sdk/ReferenceFullscreenActivity.kt

```diff
@@ -149,6 +175,50 @@ class ReferenceFullscreenActivity : AppCompatActivity() {
                 }
             })
 
+            if (adType == ReferenceFullscreenAd.ReferenceFullscreenAdFormat.REWARDED_INTERSTITIAL) {
+                val timerView = binding.referenceFullscreenTimerview
+                val closeButton = binding.referenceFullscreenClosebutton
+                val params = FrameLayout.LayoutParams(
+                    FrameLayout.LayoutParams.WRAP_CONTENT,
+                    FrameLayout.LayoutParams.WRAP_CONTENT
+                ).apply {
+                    gravity = Gravity.TOP or Gravity.END
+                    setMargins(0, 0, 10, 0)
+                }
+
+                var timer: CountDownTimer? = null
+
+                closeButton.setOnClickListener {
+                    timer?.cancel()
+
+                    if (remainingTimeMillis >= 1000) {
+                        AlertDialog.Builder(this@ReferenceFullscreenActivity)
```

Comment:
does this dialog pause the countdown? you're cancelling then starting again. does this restart it to 5s?

Reply:
There's no "pausing" with timer that I'm aware of. You cancel it, keep track of how many seconds are left, then make a new timer that starts from that to simulate pause and resume.